### ISSUE TITLE
YieldFromCanBeUsedInspection false positive

### ISSUE THREAD

**Issue by funivan** [kalessil/phpinspectionsea#1167]
First, thank you for reporting a bug or making a request. That takes time and we appreciate that!

Let's start with common details:
 
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | Bug/False-Positive          |
| Plugin         | Php Inspections (EA Ultimate) v2.0.11 |
| Language level | PHP 7.1 |

Current behaviour
```php
class CompositeLetters implements \IteratorAggregate {
    /** @var Letters[] */
    private $items;
    public function __construct(Letters ...$letters) {
      $this->items = $letters;
    }
    public function getIterator() {
      foreach ($this->items as $item) { // <-- error goes here
        yield from $item;
      }
      //After QF we get WRONG code
      //yield from $this->items;
    }
  }
  class Letters implements IteratorAggregate {
    public function getIterator() {
      yield from ['a', 'b'];
    }
  }
  foreach (new CompositeLetters(new Letters(), new Letters()) as $l) {
    echo $l . "\n";
  }
```
Invoke QF:
![screenshot from 2019-01-16 10-30-59](https://user-images.githubusercontent.com/425208/51236310-80da3780-197a-11e9-84d3-bc092da50ebe.png)
Current behavior:
--
Inspection warning

Expected behavior:
--
No inspection warning

**Comment by kalessil** [comment]
I'm lacking brain power today, @funivan can you please explain to me why we can not properly handle an array of iterators (or it's an array of generators, or probably iterators should not return generators). I'm lost.

**Comment by kalessil** [comment]
Fixed!
